lib/spack/spack/test/util/elf.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import io
+
+import pytest
+
+import llnl.util.filesystem as fs
+
+import spack.platforms
+import spack.util.elf as elf
+import spack.util.executable
+
+
+# note that our elf parser is platform independent... but I guess creating an elf file
+# is slightly more difficult with system tools on non-linux.
+def skip_unless_linux(f):
+    return pytest.mark.skipif(
+        str(spack.platforms.real_host()) != "linux",
+        reason="implementation currently requires linux",
+    )(f)
```

Comment:
What, no BSD and Illumos support? 😛 Really I think this is fine, but it makes me think we might want to have a "platform using elf" rather than "is_linux" check that we can use for things like this.

Reply:
Agreed, all you need is a compiler/linker that can output elf files... But for now linux is fine I guess